cpp/tests/io/json_test.cpp

```diff
@@ -941,4 +942,30 @@ TEST_F(JsonReaderTest, JsonExperimentalBasic)
                                  cudf::test::strings_column_wrapper({"1.1", "2.2"}));
 }
 
+TEST_F(JsonReaderTest, JsonExperimentalLines)
```

Comment:
cool test :)
I think we should also include a Python test with nested columns. I can help write such test(s), just let me know.

Reply:
Added a two python tests too. Planning to add more once we also cover type inference and type casting